src/main/scala/com/gu/googleauth/actions.scala

```diff
@@ -66,21 +67,28 @@ object AuthAction {
   */
 class AuthAction[A](val authConfig: GoogleAuthConfig, loginTarget: Call, bodyParser: BodyParser[A])(implicit val executionContext: ExecutionContext)
   extends ActionBuilder[AuthAction.UserIdentityRequest, A]
-    with ActionRefiner[Request, AuthAction.UserIdentityRequest]
     with UserIdentifier {
 
-  override protected def refine[A](request: Request[A]): Future[Either[Result, AuthAction.UserIdentityRequest[A]]] =
+  protected def refine[B](request: Request[B]): Future[Either[Result, AuthAction.UserIdentityRequest[B]]] =
```

Comment:
IntelliJ raises a warning against calling the parameter `A` - it says it's a suspicious-shadowing of the name of the class-level `A` type-parameter - so I changed the name of the method-level type-parameters to `B`.